ddln_imaginator.proto

```diff
@@ -21,6 +21,64 @@ enum ImageFormat {
     GRAY = 4;
 }
 
+message Environment {
+    message Clouds {
+        enum Type {
+            NONE = 0;
+            ALTOCUMULUS = 1;
+            CLOUDS_BASE = 2;
+            CUMULUS = 3;
+            NIMBOSTRATUS = 4;
+            STRATOCUMULUS = 5;
+            STRATUS = 6;
+        }
+        optional Type type = 1;
+        optional float altitude_m = 2;
+        optional Point3d velocity = 3; // Only use the directional x,y,z values to set the wind in the Engine.
+    }
+    optional Clouds clouds = 1;
+
+    message Haze {
+        enum Type {
+            DEFAULT = 0; // Visibility = 20000, density = 1
+            CUSTOM = 1; // Custom visibility and density as specified below
+        }
+        optional Type type = 1;
+        optional float visibility = 2; // recommended 0 - 20000
+        optional float density = 3; // recommended 0.001 - 100
+    }
+    optional Haze haze = 2;
+
+    message Fog {
+        optional bool enabled = 1;
+        optional float multiplier = 2 ; // recommended 0.01 - 0.0001
+    }
+    optional Fog fog = 3;
+
+    message Sun {
+        enum Type {
+            DEFAULT = 0;
+            CUSTOM = 1;
+        }
+        optional Type type = 1;
+        optional float azimuth_deg = 2;
+        optional float altitude_deg = 3; // should not exceed 90 degree.
+        optional float intensity = 4; // scaled 0 - 1
+    }
+    optional Sun sun = 4;
+
+    message Precipitation {
+        enum Type {
+            NONE = 0;
+            RAIN = 1;
+            SNOW = 2;
+        }
+        optional Type type = 1;
+        optional float intensity = 2; // scaled 0 - 1
```

Comment:
We also need velocity here, relative to the camera.
Imaginator is stateless, so it doesn't have a camera velocity signal, but precipitation looks very different at speed.

Reply:
For now, snow falls with 6m/s and rain with 20m/s in engine time.
Option is we can set the rendering duration between the updates(time between two frames). If we have the camera fps, we can try to match them. But the risk is that the fixed duration is not always reliable(depending on the performance I guess). It's the same issue with the cloud.